src/createServer.js

```diff
@@ -1,7 +1,356 @@
+    });
+
+    return res.status(201).json(newExpense);
+  } catch (error) {
+    return res
+      .status(400)
+      .json({ message: 'Error creating expense', error: error.message });
+  }
+});
+
+router.patch('/expenses/:id', async (req, res) => {
+  const id = Number(req.params.id);
+  const { title, amount, category, note, spentAt } = req.body;
+
+  try {
+    const expense = await Expense.findByPk(id);
+
+    if (!expense) {
+      return res.status(404).json({ message: 'Expense not found' });
+    }
+
+    let updatedAmount = amount;
+
+    if (amount !== undefined) {
+      updatedAmount = parseFloat(amount);
+
+      if (isNaN(updatedAmount)) {
+        return res.status(400).json({ message: 'Invalid amount format' });
+      }
+      updatedAmount = updatedAmount.toFixed(2);
+    }
+
+    await expense.update({
+      title,
+      amount: updatedAmount,
+      category,
+      note,
+      spentAt,
+    });
+
+    await expense.reload();
+
+    res.status(200).json(expense);
+  } catch (error) {
+    res.status(500).json({ message: 'Error updating expense' });
+  }
+});
+
+router.delete('/expenses/:id', async (req, res) => {
+  const id = Number(req.params.id);
+
+  try {
+    const expense = await Expense.findByPk(id);
+
+    if (!expense) {
+      return res.status(404).json({ message: 'Expense not found' });
+    }
+
+    await expense.destroy();
+    res.sendStatus(204);
+  } catch (error) {
+    res.status(500).json({ message: 'Error deleting expense' });
+  }
+});
+
+router.get('/users', async (req, res) => {
+  try {
+    const users = await User.findAll();
+
+    res.json(users);
+  } catch (error) {
+    res.status(500).json({ message: 'Error fetching users' });
+  }
+});
+
+router.get('/users/:id', async (req, res) => {
+  const id = Number(req.params.id);
+
+  try {
+    const user = await User.findByPk(id);
+
+    if (!user) {
+      return res.status(404).json({ message: 'User not found' });
+    }
+
+    res.json(user);
+  } catch (error) {
+    res.status(500).json({ message: 'Error fetching user' });
+  }
+});
+
+router.post('/users', async (req, res) => {
+  const { name } = req.body;
+
+  try {
+    if (!name) {
+      return res.status(400).json({ message: 'Name is required' });
+    }
+
+    const newUser = await User.create({ name });
+
+    res.status(201).json(newUser);
+  } catch (error) {
+    res.status(500).json({ message: 'Error creating user' });
+  }
+});
+
+router.patch('/users/:id', async (req, res) => {
+  const id = Number(req.params.id);
+  const { name } = req.body;
+
+  try {
+    const user = await User.findByPk(id);
+
+    if (!user) {
+      return res.status(404).json({ message: 'User not found' });
+    }
+
+    if (name === undefined || name.trim() === '') {
+      return res.status(400).json({ message: 'Name is required' });
+    }
+
+    await user.update({ name });
+    res.status(200).json(user);
+  } catch (error) {
+    res.status(500).json({ message: 'Error updating user' });
```

Comment:
Consider including the error message in the response to provide more context for debugging.